flocker/ca/_ca.py

```diff
@@ -214,44 +214,67 @@ def flocker_keypair():
     )
 
 
-def load_certificate_from_path(path, key_filename, cert_filename):
+def load_certificate_file(path):
     """
-    Load a certificate and keypair from a specified path.
+    Load a certificate from a specified path.
 
-    :param FilePath path: Directory where certificate and key files
-        are stored.
-    :param bytes key_filename: The file name of the private key.
-    :param bytes cert_filename: The file name of the certificate.
+    :param FilePath path: Absolute path to certificate file.
 
-    :return: A ``tuple`` containing the loaded key and certificate
-        instances.
+    :return: A ``Certificate`` instance representing the parsed
+        certificate data.
     """
-    certPath = path.child(cert_filename)
-    keyPath = path.child(key_filename)
-
     try:
-        certFile = certPath.open()
+        certificate_file = path.open()
     except IOError as e:
         code, failure = e
         raise PathError(
             b"Certificate file could not be opened.",
             e.filename, code, failure
         )
+    certificate = Certificate.load(
+        certificate_file.read(), format=crypto.FILETYPE_PEM)
+    return certificate
 
+
+def load_key_file(path):
+    """
+    Load a private key from a specified path.
+
+    :param FilePath path: Absolute path to certificate file.
+
+    :return: A ``ComparableKeyPair`` instance representing the parsed
+        key data.
+    """
     try:
-        keyFile = keyPath.open()
+        key_file = path.open()
     except IOError as e:
         code, failure = e
         raise PathError(
             b"Private key file could not be opened.",
             e.filename, code, failure
         )
-
-    certificate = Certificate.load(
-        certFile.read(), format=crypto.FILETYPE_PEM)
     keypair = ComparableKeyPair(
-        keypair=KeyPair.load(keyFile.read(), format=crypto.FILETYPE_PEM)
+        keypair=KeyPair.load(key_file.read(), format=crypto.FILETYPE_PEM)
     )
+    return keypair
+
+
+def load_certificate_from_path(path, key_filename, cert_filename):
+    """
+    Load a certificate and keypair from a specified path.
+
+    :param FilePath path: Directory where certificate and key files
+        are stored.
+    :param bytes key_filename: The file name of the private key.
+    :param bytes cert_filename: The file name of the certificate.
+
+    :return: A ``tuple`` containing the loaded key and certificate
```

Comment:
We do return tuples in some places. Note that the return value of this method is not a behaviour that changes in this branch.